buildenv/jenkins/jobs/infrastructure/copyrightCheck.groovy

```diff
@@ -21,79 +21,104 @@
-                                            depth: 0,
-                                            noTags: false,
-                                            reference: '/home/jenkins/openjdk_cache',
-                                            shallow: false]],
-                            userRemoteConfigs: [[refspec: "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/* +refs/heads/${ghprbTargetBranch}:refs/remotes/origin/${ghprbTargetBranch}",
-                                                    url: SRC_REPO]]]
-                FILES = sh (
-                    script: "git diff -C --diff-filter=ACM --name-only origin/${ghprbTargetBranch} HEAD",
-                    returnStdout: true
-                ).trim()
-                echo FILES
-                if (FILES == "") {
-                    echo "There are no files to check for copyrights"
-                } else {
-                    def FILES_LIST = FILES.split("\\r?\\n")
-                    DATE_YEAR = sh (
-                        script: "date +%Y",
-                        returnStdout: true
-                    ).trim()
+                try {
+                    def remoteConfigParameters = [refspec: "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/* +refs/heads/${ghprbTargetBranch}:refs/remotes/origin/${ghprbTargetBranch}",
+                                                      url: SRC_REPO]
+                    if (GIT_CREDENTIALS_ID) {
+                        remoteConfigParameters.put("credentialsId", "${GIT_CREDENTIALS_ID}")
+                    }
+                    String REPO_CACHE_DIR = params.REPO_CACHE_DIR ?: "${env.HOME}/openjdk_cache"
 
-                    // Set a different Copyright regex depending on the Repo the PR is from
-                    REGEX = "\'Copyright \\(c\\) ([0-9]{4}), ${DATE_YEAR}\'"
+                    checkout changelog: false, poll: false,
+                            scm: [$class: 'GitSCM',
+                                branches: [[name: sha1]],
+                                doGenerateSubmoduleConfigurations: false,
+                                extensions: [[$class: 'CloneOption',
+                                                depth: 0,
+                                                noTags: false,
+                                                reference: "${REPO_CACHE_DIR}",
+                                                shallow: false]],
+                                userRemoteConfigs: [remoteConfigParameters]]
 
-                    IGNORE_LIST = []
-                    if (fileExists("${WORKSPACE}/.copyrightignore")) {
-                        IGNORE_LIST = readFile '.copyrightignore'
-                        IGNORE_LIST = IGNORE_LIST.tokenize("\n")
+                    if (GIT_CREDENTIALS_ID) {
+                        sshagent(credentials:["${GIT_CREDENTIALS_ID}"]) {
+                            FILES = get_files()
```

Comment:
Does the `git diff` command actually reach out to the remote? I would have assumed it compared against the local git cache.

Reply:
Yes